modules/nf-core/seurat/multiseq/templates/MULTIseq.R

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env Rscript
+
+# Tool from Seurat released under the MIT license.
+# Ported to nf-core/modules with template by Mariana Gonzales Andre
+
+#' Check for Non-Empty, Non-Whitespace String
+#'
+#' This function checks if the input is non-NULL and contains more than just whitespace.
+#' It returns TRUE if the input is a non-empty, non-whitespace string, and FALSE otherwise.
+#' @param input A variable to check.
+#' @return A logical value: TRUE if the input is a valid, non-empty, non-whitespace string; FALSE otherwise.
+#' @examples
+#' is_valid_string("Hello World") # Returns TRUE
+#' is_valid_string("   ")         # Returns FALSE
+#' is_valid_string(NULL)          # Returns FALSE
+
+is_valid_string <- function(input) {
+    !is.null(input) && nzchar(trimws(input))
+}
+
+parse_args <- function(x){
+    args_list <- unlist(strsplit(x, ' ?--')[[1]])[-1]
+    args_vals <- lapply(args_list, function(x) scan(text=x, what='character', quiet = TRUE))
+
+    # Ensure the option vectors are length 2 (key/ value) to catch empty ones
+    args_vals <- lapply(args_vals, function(z){ length(z) <- 2; z})
+
+    parsed_args <- structure(lapply(args_vals, function(x) x[2]), names = lapply(args_vals, function(x) x[1]))
+    parsed_args[! is.na(parsed_args)]
+}
+
+################################################
+################################################
+## PARSE PARAMETERS FROM NEXTFLOW             ##
+################################################
+################################################
+
+# Set defaults and classes
+
+opt <- list(
+    output_prefix = ifelse('$task.ext.prefix' == 'null', '$meta.id', '$task.ext.prefix'),
+    hto_matrix = '$hto_matrix',
+    rna_matrix = '$rna_matrix',
+    produce_plots = '$produce_plots'
+)
+
+opt_types <- lapply(opt, class)
+
+# Apply parameter overrides
+
+args_opt <- parse_args('$task.ext.args')
+
+# All values from ext.args are stored as strings
+# Function to transform strings to the correct class
+convert_element <- function(x) {
```

Comment:
Could you move the functions to the top, with the other function? Would help with readability.

Also, add the proper roxygen-style function documentation to all.

Reply:
I totally agree with you on having a wrapper as thin as possible, I'll check if some parts of the CLI can be applied here, I also thought from the beginning that the block of code for the plots could be too large for the module, however splitting the code would imply adding an "intermediate layer" in which we would obtain the results from demultiplexing as an RDS object and produce plots, the problem is that not all plots included in the tool are available under the CLI